src/coverage.rs

```diff
@@ -0,0 +1,45 @@
+use std::env;
+use std::fs::OpenOptions;
+use std::io::Write;
+use std::ops::Range;
+use std::sync::Mutex;
+use std::collections::HashMap;
+
+lazy_static! {
+    static ref COVERAGE_ALREDY_CHECKED: Mutex<HashMap<usize, ()>> = Mutex::new(HashMap::new());
+}
+
+pub fn report_coverage(mutations: Range<usize>) {
+    let mutagen_coverage = env::var_os("MUTAGEN_COVERAGE");
+    if mutagen_coverage.is_none() {
+        return
+    }
+
+    match COVERAGE_ALREDY_CHECKED.lock() {
```

Comment:
This is probably funny, but I find this to be a bit heavy. :smile:

We could generate one static `AtomicBool` per mutation and use that.

Reply:
Thanks for the feedback, adressed on: 648be7c3a7065b9aa9202eb28b73960aa03ba810

I've needed an extra env var to be able to create a vector which is long enough. If there's any alternative, let me know.